packages/cli/src/task-runners/task-managers/task-requester.ts

```diff
@@ -71,6 +75,15 @@ export abstract class TaskRequester {
 		private readonly errorReporter: ErrorReporter,
 	) {}
 
+	setRunnerUnavailable(taskType: string, reason: string) {
+		this.unavailableRunners.set(taskType, reason);
+	}
+
+	getRunnerStatus(taskType: string): RunnerStatus {
```

Comment:
More type-safe to use a union `'javascript' | 'python'` but `startJob` so far has been using simply `string` and this cascades in multiple spots, so that'd be a bigger refactor.